core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -684,6 +685,18 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(new RemoveRedundantIdentityProjections())),
 
+                new IterativeOptimizer(
```

Comment:
We should try to place this in one of the existing IterativeOptimizers instead of adding a new one unless absolutely necessary.

Reply:
Let's add this where we do other pushdowns (which is more than once).
Also, we should make sure to attempt this before ReorderJoins